crates/federate/src/worker.rs

```diff
@@ -711,15 +709,16 @@ mod test {
     let rcv = data.stats_receiver.recv().await.unwrap();
     assert_eq!(3, rcv.state.fail_count);
 
-    // now make sends successfu
+    // now make sends successful
     *data.respond_with_error.write().unwrap() = false;
 
     // give it time for retry and to deliver activities
-    sleep(Duration::from_secs(5)).await;
+    sleep(Duration::from_secs(2)).await;
 
-    // fail count goes back to 0
-    let rcv = data.stats_receiver.recv().await.unwrap();
-    assert_eq!(0, rcv.state.fail_count);
+    // fail count goes back to 0 (need to skip all intermediary stats and get the final one)
+    let mut buffer = vec![];
+    data.stats_receiver.recv_many(&mut buffer, 10).await;
+    assert_eq!(0, buffer.last().unwrap().state.fail_count);
```

Comment:
This was implemented wrong, and somehow it started to fail consistently with this PR. So I made a few changes.